### ISSUE TITLE
[Feature Request]: disabling anti-blur with an UI setting.

### ISSUE THREAD

**Issue by gilderoylockhart01** [PCSX2/pcsx2#5479]
### Description

PCSX2 comes with a built-in anti-blur algo which works well enough, but it may not be of the liking of everyone as it could kill some intended effects which renders fine at native resolution.

The request here is pretty straightforward. Disabling the anti blur with an UI button in the GS settings window.

With the incoming QT GUI it should be doable to do such an option.

### Reason

Accuracy. Some peoples may want the blur to be active.

### Examples

Devil may cry does tend to use that quite often and this looks good at native, why getting rid of that?

**Comment by lightningterror** [comment]
A better solution to remove the code in itself since it is a hack.

**Comment by refractionpcsx2** [comment]
I don't think we'll hear the end of it if we removed it completely.

**Comment by lightningterror** [comment]
What about disabling it on native res, or any other way making it suck less, otherwise I am against adding more hack gui options.

**Comment by refractionpcsx2** [comment]
Not sure about disabling it on native res, but disabling it on software mode makes sense, since that's the "accurate" option.

**Comment by gilderoylockhart01** [comment]
Well, in the GUI there are 2 hacks that can be removed.

Auto flush is more of a rendering behaviour than a hack if I am not wrong and memory wrapping should be enabled all the time since it's more accurate, that leaves us with 2 less entries.

The problem with the software only approach is the speed, if we do that only on software it's going to be a nightmare in that regard.

And also, disabling it with upscaled resolution might not be a good idea too if the effect still looks good.

**Comment by Mrlinkwii** [comment]
>The problem with the software only approach is the speed, if we do that only on software it's going to be a nightmare in that regard.

in this  regard it has gotten better recently  due to prs that have been merged , so its not a big issue as before

**Comment by RedDevilus** [comment]
Memory Wrapping + Preload Frame Data does look promising to be incorporated. Ironically they are called hacks when it's more accurate than the default behavior.

**Comment by gilderoylockhart01** [comment]
Preload frame data should deffinitely *not* be included in it's current state ^^. Some games like FF12 hate that.

**Comment by refractionpcsx2** [comment]
yeah preload frame would need checking if the bugs can be fixed before we turned it permanently on.  auto flush should be a general option since it seems to interfere between SW and HW anyway, it seems.  Memory Wrapping we will need to test, but yeah everything else needs to stay there.

**Comment by AaronBPaden** [comment]
Oh yes, if such a thing exists I'd definitely like an option to remove it that doesn't involve using software rendering. I play at native res and don't want my games to be "enhanced".

WRT UI there could just be a checkbox either in the renderer or shader tab, whichever makes more sense. Or there could be a new tab a la Dolphin called "Enhancements" where that and similar things go.

**Comment by gilderoylockhart01** [comment]
Actually, it would be nice to maybe do a PR for integrating the VU I-BIT, VU ADD/SUB and memory wrapping directly to the core. That will needs testing, I can handle a good part of it and that will free us of some hacks which are just, not hacks.

For memory wrapping I did some tests already and it matches the software render without any drawbacks in the games I tested, ditto for booth VU """""hacks""""" but the latter will obviously need more testing than I did.
